File: Austin Page - 100725236/Python code (local)/producer.py
import json
from kafka import KafkaProducer


def serializer(message):
    return json.dumps(message).encode('utf-8')


producer = KafkaProducer(bootstrap_servers="localhost:9093",
                         value_serializer=serializer)

if __name__ == '__main__':
    while True:
        try:
            message = input("Message: ")
            topic = input("Topic: ")
            part = input("Partition: ")
            producer.send(str(topic), value=str(message), partition=int(part))
            print(" ---------- Message sent! ---------- ")
        except:
            print(" ---------- An error occurred ---------- ")

File: Saood Imran - 100620420/node-js/producer.js
const {Kafka} = require("kafkajs")
const msg = process.argv[2];
run();
async function run(){ 
    try {
        const kafka = new Kafka({
            "clientID": "myapp",
            "brokers": ["localhost:9093", "localhost:9094", "localhost:9095"]
        })

        const producer = kafka.producer();
        console.log("Connecting..")
        await producer.connect()
        console.log("Connected!")
        const partition = msg[0] < "N" ? 0 : 1
        const result = await producer.send({
            "topic": "Users",
            "messages": [
                {
                    "value": msg,
                    "partition": partition
                }
            ]
        })
        console.log(`Sent message ${JSON.stringify(result)}`)
        await producer.disconnect();
    } catch (error) {
        console.error(`Error occured ${ex}`)
    }

    finally{
        process.exit();
    }
}